db/schema.sql

```diff
@@ -1,33 +1,45 @@
-/*  If you add/change anything, update
-    boinc_db.cpp,h
-    and if needed:
-    py/Boinc/
-        database.py
-    html/
-        inc/
-            host.inc (host)
-            db_ops.inc
-        ops/
-            db_update.php
-        user/
-            create_account_action.php (user)
-            team_create_action.php (team)
-    sched/
-        db_dump.cpp (host, user, team)
-        db_purge.cpp (workunit, result)
+/*
+    If you add/change tables:
+    - if used by C++ code, update
+        db/
+            boinc_db.cpp,h
+            boinc_db_types.h
+        sched/
+            db_dump.cpp (host, user, team)
+            db_purge.cpp (workunit, result)
+    - if used by Python scripts (make_project, update_versions), update
+        py/Boinc/database.py
+    - if used by PHP code, update as needed
+        html/
+            inc/
+                host.inc (host)
+                db_ops.inc
+            ops/
+                db_update.php
+            user/
+                create_account_action.php (user)
+                team_create_action.php (team)
 */
-/* Fields are documented in boinc_db.h */
-/* Do not replace this with an automatically generated schema */
 
-/* type is specified as InnoDB for most tables.
-   Supposedly this gives better performance.
-   The others (post, thread, profile) are myISAM
-   because it supports fulltext index
-*/
+-- Most fields are documented in boinc_db_types.h
 
-/* fields ending with id (but not _id) are treated specially
-   by the Python code (db_base.py)
-*/
+-- All fields should be not null
+-- Fields should generally have a default
+-- (newer MySQL versions don't have automatic defaults)
+
+-- add new fields to the end of the table
+-- (makes it easier to update C++ code)
+
+-- Engine is specified as InnoDB for most tables;
+-- supposedly this gives better performance.
+-- Some (post, thread, profile) are myISAM because it supports fulltext index
+
+-- Going forward, use double for unix times (no 32-bit problem)
+
+-- Put index definitions in constraints.sql, not here
+
+-- fields ending with id (but not _id) are treated specially
+-- by the Python code (db_base.py)
```

Comment:
Do you mind adding some comments about how they are treated specially?   This doesn't tell me what I need to do in order to properly follow convention.